app/assets/javascripts/discourse/app/controllers/delete-topic-confirm.js

```diff
@@ -0,0 +1,37 @@
+import I18n from "I18n";
+import discourseComputed from "discourse-common/utils/decorators";
+import Controller, { inject } from "@ember/controller";
+import ModalFunctionality from "discourse/mixins/modal-functionality";
+
+// Modal that displays confirmation text when user deletes a topic
+// The modal will display only if the topic exceeds a certain amount of views
+export default Controller.extend(ModalFunctionality, {
+  topicController: inject("topic"),
+
+  @discourseComputed("deletingTopic")
+  buttonTitle(deletingTopic) {
+    return deletingTopic
+      ? I18n.t("deleting")
+      : I18n.t("post.controls.delete_topic_confirm_modal_yes");
+  },
+
+  onShow() {
+    this.set("deletingTopic", false);
+  },
+
+  actions: {
```

Comment:
can you use `@action` please ? you will see examples at other places in the codebase

Reply:
Will do, thank you